src/table/use-column-widths.tsx

```diff
@@ -16,25 +16,19 @@ export interface ColumnWidthDefinition {
   width?: string | number;
 }
 
-function readWidths(
+function readWidth(
   getCell: (columnId: PropertyKey) => null | HTMLElement,
-  visibleColumns: readonly ColumnWidthDefinition[]
+  column: ColumnWidthDefinition,
+  isLastColumn: boolean
 ) {
-  const result = new Map<PropertyKey, number>();
-  for (let index = 0; index < visibleColumns.length; index++) {
-    const column = visibleColumns[index];
-    let width = (column.width as number) || 0;
-    const minWidth = (column.minWidth as number) || width || DEFAULT_COLUMN_WIDTH;
-    if (
-      !width && // read width from the DOM if it is missing in the config
-      index !== visibleColumns.length - 1 // skip reading for the last column, because it expands to fully fit the container
-    ) {
-      const colEl = getCell(column.id);
-      width = colEl ? getLogicalBoundingClientRect(colEl).inlineSize : DEFAULT_COLUMN_WIDTH;
-    }
-    result.set(column.id, Math.max(width, minWidth));
+  let width = (treatAsNumber(column.width) && (column.width as number)) || 0;
+  const minWidth = (treatAsNumber(column.minWidth) && (column.minWidth as number)) || width || DEFAULT_COLUMN_WIDTH;
+  // read width from the DOM if it is missing in the config
+  if (!width && !isLastColumn) {
+    const colEl = getCell(column.id);
+    width = colEl ? getLogicalBoundingClientRect(colEl).inlineSize : DEFAULT_COLUMN_WIDTH;
```

Comment:
If I understood correctly, we call readWidth on each column to compare it with the prev state to know if the configuration changed, like a new column was added or its width/minWidth/maxWidth change. In that case, would it be possible to avoid reading the actual element width?